tasks/mpi/kalinin_d_matrix_mult_hor_a_vert_b/func_tests/main.cpp

```diff
@@ -0,0 +1,374 @@
+#include <gtest/gtest.h>
+
+#include <boost/mpi/communicator.hpp>
+#include <boost/mpi/environment.hpp>
+#include <random>
+#include <vector>
+
+#include "mpi/kalinin_d_matrix_mult_hor_a_vert_b/include/ops_mpi.hpp"
+
+namespace kalinin_d_matrix_mult_hor_a_vert_b_mpi {
+void get_random_matrix(std::vector<int> &mat) {
+  std::random_device dev;
+  std::mt19937 gen(dev());
+
+  for (size_t i = 0; i < mat.size(); ++i) {
+    mat[i] = gen() % 10 - 5;
+  }
+}
+
+}  // namespace kalinin_d_matrix_mult_hor_a_vert_b_mpi
+
+TEST(kalinin_d_matrix_mult_hor_a_vert_b_mpi, InitializationWithEmptyInputs) {
+  boost::mpi::communicator world;
+  std::vector<int> global_A;
+  std::vector<int> global_B;
+  std::vector<int> global_res;
+
+  std::shared_ptr<ppc::core::TaskData> taskDataPar = std::make_shared<ppc::core::TaskData>();
+
+  if (world.rank() == 0) {
+    taskDataPar->inputs.emplace_back(reinterpret_cast<uint8_t *>(global_A.data()));
+    taskDataPar->inputs_count.emplace_back(0);
+    taskDataPar->inputs_count.emplace_back(0);
+
+    taskDataPar->inputs.emplace_back(reinterpret_cast<uint8_t *>(global_B.data()));
+    taskDataPar->inputs_count.emplace_back(0);
+    taskDataPar->inputs_count.emplace_back(0);
+
+    taskDataPar->outputs.emplace_back(reinterpret_cast<uint8_t *>(global_res.data()));
+    taskDataPar->outputs_count.emplace_back(global_res.size());
+  }
+}
+
+TEST(kalinin_d_matrix_mult_hor_a_vert_b_mpi, InvalidTaskWithPartialInputs) {
+  boost::mpi::communicator world;
+  std::vector<int> global_A;
+  std::vector<int> global_B;
+  std::vector<int> global_res;
+
+  std::shared_ptr<ppc::core::TaskData> taskDataPar = std::make_shared<ppc::core::TaskData>();
+
+  if (world.rank() == 0) {
+    global_A.resize(100, 0);
+
+    taskDataPar->inputs.emplace_back(reinterpret_cast<uint8_t *>(global_A.data()));
+    taskDataPar->inputs_count.emplace_back(25);
+    taskDataPar->inputs_count.emplace_back(4);
+
+    taskDataPar->inputs.emplace_back(reinterpret_cast<uint8_t *>(global_B.data()));
+    taskDataPar->inputs_count.emplace_back(0);
+    taskDataPar->inputs_count.emplace_back(0);
+
+    taskDataPar->outputs.emplace_back(reinterpret_cast<uint8_t *>(global_res.data()));
+    taskDataPar->outputs_count.emplace_back(global_res.size());
+  }
+
+  auto taskParallel = std::make_shared<kalinin_d_matrix_mult_hor_a_vert_b_mpi::TestMPITaskParallel>(taskDataPar);
+  if (world.rank() == 0) {
+    EXPECT_FALSE(taskParallel->validation());
+  } else {
+    EXPECT_TRUE(taskParallel->validation());
```

Comment:
Проводите валидацию только на 0 воркере, для остальных можно поставить заглушку EXPECT_TRUE(true);
```suggestion
    EXPECT_TRUE(true);
```